src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecutionTest.java

```diff
@@ -382,6 +384,53 @@ private static void assertStepExecutions(FlowExecution e, String... steps) throw
         }
     }
 
+    @Test
+    public void flowExecutionListener() throws Exception {
+        story.addStep(new Statement() {
+                          @Override
+                          public void evaluate() throws Throwable {
+                              WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                              p.setDefinition(new CpsFlowDefinition("echo 'Running for listener'; semaphore 'wait'", true));
```

Comment:
Try inserting a `sleep 0` in the middle here and I am betting you will get a count > 2.